lib/PuppeteerSharp/JSHandle.cs

```diff
@@ -7,23 +7,44 @@
 
 namespace PuppeteerSharp
 {
+    /// <summary>
+    /// JSHandle represents an in-page JavaScript object. JSHandles can be created with the <see cref="Page.EvaluateExpressionHandleAsync(string)"/> and <see cref="Page.EvaluateFunctionHandleAsync(string, object[])"/> methods.
+    /// </summary>
     public class JSHandle
     {
-        private ExecutionContext _context;
-        protected readonly Session _client;
-        protected readonly ILogger _logger;
-
-        public JSHandle(ExecutionContext context, Session client, object remoteObject)
+        internal JSHandle(ExecutionContext context, Session client, object remoteObject)
         {
-            _context = context;
-            _client = client;
-            _logger = _client.Connection.LoggerFactory.CreateLogger(this.GetType());
+            ExecutionContext = context;
+            Client = client;
+            Logger = Client.Connection.LoggerFactory.CreateLogger(this.GetType());
             RemoteObject = remoteObject;
         }
 
-        public ExecutionContext ExecutionContext => _context;
+        /// <summary>
+        /// Gets the execution context.
+        /// </summary>
+        /// <value>The execution context.</value>
+        public ExecutionContext ExecutionContext { get; private set; }
```

Comment:
I don't think we need the `private set` you can set in the ctor even without it

Reply:
cool, I didn't know about that.

Reply:
Done @Meir017